sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -777,16 +780,12 @@ ProgramManager::ProgramPtr ProgramManager::build(
     LinkOpts = LinkOptions.c_str();
   }
 
-  // The Level Zero driver support for online linking currently has bugs, but
-  // we think the DPC++ runtime support is ready.  This environment variable
-  // gates the runtime support for online linking, so we can try enabling if a
-  // new driver is released before the next DPC++ release.
-  static bool EnableLevelZeroLink = std::getenv("SYCL_ENABLE_LEVEL_ZERO_LINK");
-  if (!EnableLevelZeroLink) {
-    if (Context->getPlugin().getBackend() == backend::level_zero) {
-      LinkDeviceLibs = false;
-    }
-  }
```

Comment:
Online linking was disabled for Level Zero only. Now it's disabled for each and every backend. Why is that?
Another solution is to enable it for OpenCL backend.

Reply:
Hi, @s-kanaev 
There are 2 reasons:
1. The device library shouldn't depend on one specific backend to work, our final goal is to make device libraries to have same behavior in all backend. So, we switch to offline link for all backends.

2. Simplifying the usage requires us to make some device libraries to be linked in by default in offline build phase. If we enable devicelib for OCL backend, we must only link "wrapper" obj device libs offline. 
However, to make L0 to work, we must link both "wrapper" and "fallback" libraries offline. Unfortunately, we can't know which backend will be used in offline build, so we have to disable all backends and switch back to "online" link transparently when L0 is ready.
Thank you very much.

Reply:
It looks like we are reaching the limit of a system if at the end everybody has to align on the intersection of all the features of the back-ends, which converges to the... empty set!
And if the root cause is a political back-end that nobody cares except the people who invented it, I am not super excited about this story jeopardizing all the other back-ends... :-/

Reply:
I agree with @keryell that special-casing for particular BEs in the generic SYCL runtime is something which should be avoided. The supposed way to avoid this is to extract the diverging piece of code into a separate function and make it part of the PI interface, hiding all specifics inside the plugin. @jinge90 - I'd suggest to add a TODO at least and maybe file an issue.

Reply:
Note that this PR **removes** the code that special-cases the L0 plugin.

Reply:
Upon second look, it seems nothing level zero-specific is in the patch. LinkDeviceLibs have existed before. But still, would be good to have TODO/issue for BE-specific stuff.

Reply:
Hi, @kbobrovs @keryell and @gmlueck 
I think all backends including Level zero will support online link, this patch is a temporary solution to make device library to work on all backends with same behaviors. Once all backends are ready, we will switch to "online" link transparently. I will add a "TODO" in another PR.
Thank you very much.